src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,58 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_SIZE = 10;
+    private final K [] keys;
+    private final V [] values;
+    private int size;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[MAX_SIZE];
+        values = (V[]) new Object[MAX_SIZE];
+    }
+
     @Override
     public void put(K key, V value) {
+        int index = getIndex(key);
+        if (index
+                != -1 || isFull()) {
+            values[index] = value;
+            return;
+        }
+        if (size < MAX_SIZE) {
```

Comment:
you can reuse your code 
```suggestion
        } else if (!isFull()) {
```